File: 2.ödev/ColonyClash/src/BUretim.c
/**
*
* @author Sena Nur Erdem     sena.erdem1@ogr.sakarya.edu.tr
* @since 5 Mayıs 2023
*/
#include "BUretim.h"
#include <stdlib.h>

int buretim_uret(Uretim *uretim) {
    return 2 * (rand() % 5) + 2; // 1 ile 10 arası çift sayı üretir
}

Uretim *buretim_olustur() {
    return uretim_olustur(buretim_uret);
}

File: 3.ödev/Colony/src/colony/BUretim.java
/**
*
* @author Sena Nur Erdem     sena.erdem1@ogr.sakarya.edu.tr
* @since 23 Mayıs 2023
* <p>
*  Rastgele bir çift sayı üreten bir üretim sınıfı
* </p>
*/
package colony;
import java.util.Random;

//BUretim sınıfı, Uretim arayüzünü uygular
class BUretim implements Uretim {
    public int uret() {
        Random rand = new Random();
        return 2 * (rand.nextInt(5)) + 2; // 1 ile 10 arası çift sayı üretir
    }
}